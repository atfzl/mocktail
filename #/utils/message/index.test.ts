import { postMessage, receiveMessage } from '.';

describe('postMessage', () => {
  it('default', async done => {
    const stopReceivingMessages = receiveMessage(async payload => {
      expect(payload).toMatchSnapshot();
    });

    await postMessage({ some: 'message' });

    stopReceivingMessages();

    done();
  });

  it('with a reply', async done => {
    const stopReceivingMessages = receiveMessage(async payload => {
      expect(payload).toMatchSnapshot();
      return { replyMessage: '1' };
    });

    const reply = await postMessage({ some: 'new message' });

    expect(reply).toMatchSnapshot();

    stopReceivingMessages();

    done();
  });
});
