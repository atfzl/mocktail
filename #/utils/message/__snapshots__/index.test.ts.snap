// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`postMessage default 1`] = `
Object {
  "some": "message",
}
`;

exports[`postMessage with a reply 1`] = `
Object {
  "some": "new message",
}
`;

exports[`postMessage with a reply 2`] = `
Object {
  "replyMessage": "1",
}
`;
